tools/tokamak_grids/gridgen/process_grid.pro

```diff
@@ -1619,6 +1619,45 @@ retrybetacalc:
   psi_bndry = mesh.faxis + mesh.fnorm
   s = file_write(handle, "psi_bndry", psi_bndry)
 
+  ; save some version information
+  ;
+  ; BOUT++ version information: this is set when BOUT++ is configured.
+  ; Hypnotoad doesn't require BOUT++ to have been configured, and IDL code may
+  ; have changed since 'configure' was run, so this is not 100% reliable, but
+  ; still worth saving as a sanity check
+  hypnotoad_info = ROUTINE_INFO('hypnotoad', /SOURCE)
+  hypnotoad_path = FILE_DIRNAME(hypnotoad_info.path)
+
+  ; BOUT++ git hash
+  ;; old version got git hash from bout-config, but this may be out of date
+  ;SPAWN, STRJOIN([hypnotoad_path, PATH_SEP(), '..', PATH_SEP(), '..', PATH_SEP(), '..', PATH_SEP(), 'bin/bout-config --git']), bout_git_hash, EXIT_STATUS=status
+  ;IF status THEN BEGIN
+  ;  PRINT, "WARNING: Failed to get Git hash for BOUT++, could not run '../../../bin/bout-config --git'. Have you configured BOUT++ successfully?"
+  ; new version gets current git hash directly from git-describe
+  SPAWN, STRJOIN(['cd ',hypnotoad_path, '&& git describe --always --abbrev=0 --dirty --match "NOT A TAG"']), bout_git_hash, EXIT_STATUS=status
```

Comment:
We might need to check it's a git repo first -- BOUT++ might have been installed